geomesa-gs-monitor-elasticsearch/src/main/scala/org/geomesa/gs/monitor/elastic/ExtendedRequestData.scala

```diff
@@ -0,0 +1,175 @@
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the GNU GENERAL PUBLIC LICENSE,
+ * Version 2 which accompanies this distribution and is available at
+ * https://opensource.org/licenses/GPL-2.0.
+ ***********************************************************************/
+
+package org.geomesa.gs.monitor.elastic
+
+import com.google.gson._
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.commons.lang3.StringUtils
+import org.geotools.filter.text.ecql.ECQL
+import org.geotools.geometry.jts.JTS
+import org.locationtech.geomesa.filter.FilterHelper
+import org.locationtech.jts.geom.Point
+import org.locationtech.jts.io.WKTWriter
+import org.opengis.filter.Filter
+import org.opengis.geometry.BoundingBox
+
+import java.lang.reflect.Type
+import java.nio.charset.StandardCharsets
+import java.util.Date
+import javax.naming.ldap.LdapName
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+class ExtendedRequestData(requestData: RequestData) extends RequestData(requestData) {
+
+  import org.geomesa.gs.monitor.elastic.ExtendedRequestData._
+
+  val failed: java.lang.Boolean =
+    Option(requestData.getError)
+      .isDefined
+
+  val timedOut: java.lang.Boolean =
+    Option(requestData.getError)
+      .map(_.getMessage)
+      .contains(TIMEOUT_KEY)
+
+  val bboxCentroid: Point =
+    Option(requestData.getBbox)
+      .map(JTS.toGeometry)
+      .map(_.getCentroid)
+      .orNull
+
+  @transient
+  private val queryFilter: Option[Filter] =
+    Option(requestData.getQueryString)
+      .flatMap(extractCqlFilter)
+
+  val queryAttributes: java.util.List[String] =
+    queryFilter
+      .map(FilterHelper.propertyNames)
+      .asNonEmptyJavaListOrNull
+
+  val queryCentroids: java.util.List[Point] =
+    queryFilter
+      .map(extractFilterCentroids)
+      .asNonEmptyJavaListOrNull
+
+  @transient
+  private val distinguishedName: Option[LdapName] =
+    Option(requestData.getRemoteUser)
+      .flatMap(extractDistinguishedName)
+
+  val commonNames: java.util.List[String] =
+    distinguishedName
+      .map(extractRdnValues(_, "CN"))
+      .asNonEmptyJavaListOrNull
+
+  val organizations: java.util.List[String] =
+    distinguishedName
+      .map(extractRdnValues(_, "O"))
+      .asNonEmptyJavaListOrNull
+
+  val organizationalUnits: java.util.List[String] =
+    distinguishedName
+      .map(extractRdnValues(_, "OU"))
+      .asNonEmptyJavaListOrNull
+}
+
+object ExtendedRequestData extends LazyLogging {
+
+  val CQL_FILTER_START_KEY = "CQL_FILTER="
+  val CQL_FILTER_END_KEY = "&"
+
+  val TIMEOUT_KEY = "TIMEOUT"
+
+  def getGson(excludedFields: Set[String]): Gson = {
+    new GsonBuilder()
+      .registerTypeAdapter(classOf[Point], new JsonSerializer[Point] {
+        private val WKT_WRITER = new WKTWriter
```

Comment:
is this a thread-safe class? you could probably use the  geomesa WKTUtils class which uses thread locals i believe (which makes me think this isn't thread safe...)